src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -150,7 +147,33 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
     componentDidUpdate() {
         this.updateStylesheet();
         this.updateUiViewport();
-        this.handleResize();
+
+        const {
+            style_as_list_view,
+            style_cell,
+            style_cell_conditional,
+            style_data,
+            style_data_conditional,
+            style_filter,
+            style_filter_conditional,
+            style_header,
+            style_header_conditional,
+            style_table
+        } = this.props;
+
+        this.handleResizeIf(
+            style_as_list_view,
+            style_cell,
+            style_cell_conditional,
+            style_data,
+            style_data_conditional,
+            style_filter,
+            style_filter_conditional,
+            style_header,
+            style_header_conditional,
+            style_table
+        );
```

Comment:
Only resize through React pipeline if a style prop has changed. This behaves similarly to the 🔪 `forcedResizedOnly` state prop, but also allows style changes to trigger recalculation. For example `display:none;`, changes in column widths, etc.